### PR TITLE
Magic Links

### PR DESCRIPTION
**Pull request by mrfrase3**
add UI for new login flows, along with other stuff

### CODE REVIEW

File: src/components/auth/magic/login-magic-signup.vue [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        @blur="$v.firstName.$touch()"
+      />
+      <v-text-field
+        v-model.trim="lastName"
+        :error-messages="lastNameErrors"
+        label="Last Name"
+        placeholder="Smith"
+        outlined
+        @blur="$v.lastName.$touch()"
+      />
```
Comment (eddie-atkinson) [inline src/components/auth/magic/login-magic-signup.vue:23]: We actually *might* be able to pull this info from the UWA directory. No harm in asking though

File: src/components/users/add-user-dialog.vue [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+      </v-card-title>
+      <v-card-text>
+        <v-text-field
+          v-model.trim="email"
+          label="Email"
+          outlined
+          :error-messages="emailErrors"
+          @blur="$v.email.$touch()"
+        />
+        <v-text-field
```
Comment (eddie-atkinson) [inline src/components/users/add-user-dialog.vue:18]: How is this going to work for emails to John etc?

Don't we need some kind of visitor / staff / student number to make that work even if the person doesn't have an `@...uwa.edu.au` email address

**Review by eddie-atkinson — APPROVED** [review]

File: src/components/users/add-user-dialog.vue (hunk 2 above)
Comment (mrfrase3) [reply src/components/users/add-user-dialog.vue:18]: the username is the pheme number, or whatever john needs. the email is only really for our system

File: src/components/auth/magic/login-magic-signup.vue (hunk 1 above)
Comment (mrfrase3) [reply src/components/auth/magic/login-magic-signup.vue:23]: idk, the goal was to remove dependencies on UWA systems though. And the name isn't super important.

**Pull request merged**